pkg/block/adapter.go

```diff
@@ -221,3 +220,43 @@ type Adapter interface {
 
 	RuntimeStats() map[string]string
 }
+
+type WalkerOptions struct {
+	StorageURI     *url.URL
+	SkipOutOfOrder bool
+}
+
+type WalkerWrapper struct {
+	walker Walker
+	uri    *url.URL
+}
+
+func NewWalkerWrapper(walker Walker, uri *url.URL) *WalkerWrapper {
+	return &WalkerWrapper{
+		walker: walker,
+		uri:    uri,
+	}
+}
+
+func NewWalkerWrapperFromAdapter(adapter Adapter, storageID, path string, opts WalkerOptions) (*WalkerWrapper, error) {
+	walker, err := adapter.GetWalker(storageID, opts)
+	if err != nil {
+		return nil, fmt.Errorf("creating object-store walker on path %s: %w", path, err)
+	}
+	return &WalkerWrapper{
+		walker: walker,
+		uri:    opts.StorageURI,
+	}, nil
+}
```

Comment:
It feels like to many ways to initiate the WalkerWrapper, IMO  leaving only `NewWalkerWrapper` is nicer,  `NewWalkerWrapperFromAdapter ` not worth the 1 line of code it reduces. It's a matter of taste though

Reply:
ok